ui/raidboss/raidemulator.js

```diff
@@ -0,0 +1,822 @@
+          return;
+      }
+    }
+
+    if (!result && !run)
+      return;
+
+    if (!suppressed[trigger.id])
+      suppressed[trigger.id] = [];
+
+    if (trigger.suppressSeconds) {
+      if (typeof trigger.suppressSeconds === 'function')
+        timestamp = timestamp + trigger.suppressSeconds(data, matches) * 1000;
+      else
+        timestamp = timestamp + trigger.suppressSeconds * 1000;
+      suppressed[trigger.id].push(timestamp);
+    } else {
+      timestamp = timestamp + suppress * 1000;
+      suppressed[trigger.id].push(timestamp);
+    }
+
+    let triggerElement = document.createElement('div');
+    triggerElement.setAttribute('trigger', trigger.id);
+    triggerElement.className = 'trigger';
+    triggerElement.style.left = pos + '%';
+    triggerElement.addEventListener('mouseenter', function(e) {
+      document.getElementById('tInfo').innerHTML = e.target.getAttribute('trigger');
+      document.getElementById('tInfo').style.display = 'block';
+    });
+    triggerElement.addEventListener('mouseleave', function() {
+      document.getElementById('tInfo').style.display = 'none';
+    });
+    // DEBUG:
+    // btriggerElement.setAttribute('from', data.me);
+    // triggerElement.setAttribute('time', timestamp);
+
+    document.getElementById('triggerline-' + role).appendChild(triggerElement);
+
+    return;
+  }
+};
+
+function dateFromLogLine(log) {
+  let m = log.match(/\[(\d\d):(\d\d):(\d\d).(\d\d\d)\]/);
+  if (!m)
+    return;
+  let date = new Date();
+  date.setHours(m[1]);
+  date.setMinutes(m[2]);
+  date.setSeconds(m[3]);
+  date.setMilliseconds(m[4]);
+  return date;
+}
+
+// Only listen to the import log event here and *not* the zone changed event.
+// The log collector figures out the zone from the logs itself, and not through
+// what ACT sends while importing (which races with sending logs).
+document.addEventListener('onImportLogEvent', function(e) {
+  gLogCollector.AppendImportLogs(e.detail.logs);
+});
+
+document.addEventListener('DOMContentLoaded', function() {
+  let fightListElement = document.getElementById('fight-picker');
+  let timerElements = document.getElementsByClassName('triggerline-timer');
+  let elapsedElement = document.getElementById('elapsed-time');
+  let infoElement = document.getElementById('info-panel');
+
+  let partyElement = document.getElementById('party');
+  let currentPlayerElement = document.getElementById('player');
+  let triggerInfoElement = document.getElementById('tInfo');
+
+  gEmulatorView = new EmulatorView(
+      fightListElement, timerElements, elapsedElement, infoElement,
+      partyElement, currentPlayerElement, triggerInfoElement
+  );
+  gLogCollector = new LogCollector(gEmulatorView.AddFight.bind(gEmulatorView));
+
+  document.addEventListener('onPlayerChangedEvent', function(e) {
+    document.getElementById('player').textContent = e.detail.name;
+  });
+
+  // Timeout is required because the overlay is slower and will break the Zone
```

Comment:
What does RestoreFights do that depends on the zone? If you remove the "SelectFight" part of RestoreFights, would it do the right thing?

Reply:
Actually only SelectFight does depend on the zone.
The SelectFight part happens to be in RestoreFights to keep this tool as comfortable as possible while editing trigger files. In my opinion the SelectFight part is a great part of RestoreFights but isn't vital.